OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -140,35 +156,42 @@ OptimizelyDecideOption[] options
             var ignoreUPS = Array.Exists(options,
                 option => option == OptimizelyDecideOption.IGNORE_USER_PROFILE_SERVICE);
 
-            UserProfile userProfile = null;
+
             if (!ignoreUPS && UserProfileService != null)
             {
                 try
                 {
-                    var userProfileMap = UserProfileService.Lookup(user.GetUserId());
-                    if (userProfileMap != null &&
-                        UserProfileUtil.IsValidUserProfileMap(userProfileMap))
+                    if (_userProfile == null)
```

Comment:
For legacy, we need to load fresh _userProfile on every getVariation() call (even if _userProfile not null)